addons/project/security/project_security.xml

```diff
@@ -109,7 +109,11 @@
         <field name="domain_force">[
             '&amp;',
                 ('privacy_visibility', '=', 'portal'),
-                ('message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                '|',
+                    ('message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                    '&amp;',
+                        ('allow_project_sharing', '=', True),
+                        ('project_sharing_access_ids.user_id', 'in', [user.id]),
```

Comment:
is this change necessary if the portal user is already a follower of the project?

Reply:
This rule is only applied for the read access, and the shared portal user is always follower of the project. 

So your point is valid ;)

Reply:
Indeed, it isn't necessary. We could just check if the portal user is follower of the project, since I add the portal user as follower of the project when the internal user adds an access to this portal user to project sharing feature.